Userland/Applications/Browser/Tab.cpp

```diff
@@ -393,6 +395,29 @@ Tab::Tab(BrowserWindow& window)
     };
 }
 
+Optional<URL> Tab::read_url(bool can_append_extension)
```

Comment:
`read_url` sounds like we're reading a URL from somewhere. If this is the cooked URL from the location bar, perhaps we can call it `url_from_location_bar()`.